cyral/resource_cyral_repository_user_account.go

```diff
@@ -0,0 +1,596 @@
+		return &UserAccountResource{}
+	},
+}
+
+func resourceRepositoryUserAccount() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateResource(
+			ResourceOperationConfig{
+				Name:       "RepositoryUserAccountCreate",
+				HttpMethod: http.MethodPost,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf(
+						"https://%s/v1/repos/%s/userAccounts",
+						c.ControlPlane,
+						d.Get("repository_id").(string),
+					)
+				},
+				NewResourceData: func() ResourceData {
+					return &UserAccountResource{}
+				},
+				NewResponseData: func(_ *schema.ResourceData) ResponseData {
+					return &CreateUserAccountResponse{}
+				},
+			},
+			ReadRepositoryUserAccountConfig,
+		),
+		ReadContext: ReadResource(ReadRepositoryUserAccountConfig),
+		UpdateContext: UpdateResource(
+			ResourceOperationConfig{
+				Name:       "RepositoryUserAccountUpdate",
+				HttpMethod: http.MethodPut,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf(
+						"https://%s/v1/repos/%s/userAccounts/%s",
+						c.ControlPlane,
+						d.Get("repository_id").(string),
+						d.Id(),
+					)
+				},
+				NewResourceData: func() ResourceData {
+					return &UserAccountResource{}
+				},
+			},
+			ReadRepositoryUserAccountConfig,
+		),
+		DeleteContext: DeleteResource(
+			ResourceOperationConfig{
+				Name:       "RepositoryUserAccountDelete",
+				HttpMethod: http.MethodDelete,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf(
+						"https://%s/v1/repos/%s/userAccounts/%s",
+						c.ControlPlane,
+						d.Get("repository_id").(string),
+						d.Id(),
+					)
+				},
+			},
+		),
+		Importer: &schema.ResourceImporter{
+			StateContext: func(
+				ctx context.Context,
+				d *schema.ResourceData,
+				i interface{},
+			) ([]*schema.ResourceData, error) {
+				ids, err := unmarshalComposedID(d.Id(), "/", 2)
+				if err != nil {
+					return nil, fmt.Errorf(
+						"failed to unmarshal ID: %v",
+						err,
+					)
+				}
+				repositoryID := ids[0]
+				userAccountID := ids[1]
+				err = d.Set("repository_id", repositoryID)
+				if err != nil {
+					return nil, fmt.Errorf(
+						"failed to set 'repository_id': %v",
+						err,
+					)
+				}
+				d.SetId(userAccountID)
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "ID of this resource in Cyral environment.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+
+			"repository_id": {
+				Description: "ID of the repository.",
+				Required:    true,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+			},
+
+			"name": {
+				Description: "The name of the User Account.",
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+
+			"auth_database_name": {
+				Description: "For User Accounts associated with a MongoDB cyral_repository, " +
```

Comment:
```suggestion
				Description: "For User Accounts associated with a MongoDB `cyral_repository`, " +
```

The descriptions can use markdown notation. We usually highlight all the provider resources, links and other stuff in order to make the documentation easier to follow. References to our public docs are also welcome where they make sense. :-)